go.mod

```
@@ -2,6 +2,8 @@ module github.com/buildkite/go-pipeline
 
 go 1.21.0
 
+retract v1.0.0 // Published accidentally. We'll skip this version and jump straight to v1.0.1 when we're ready to publish
```

I think for it to work we have to publish a v1.0.1 that retracts both v1.0.0 and v1.0.1

```suggestion
retract (
    v1.0.0 // Published accidentally. 
    v1.0.1 // Solely to publish the retraction of v1.0.0.
    // We'll skip straight to v1.0.2 when we're ready to publish
)
```

I believe we can still use v0.x numbering before then. i.e. if we tag v0.1.0 we can `go get ...@v0.1.0` and it should work.

---

would publishing v0.1.0 not also retract v1.0.0?

---

Having never done this myself, I can't speak from experience, but the reference says:

> To retract a version, a module author should add a retract directive to go.mod, then publish a new version containing that directive. The new version must be higher than other release or pre-release versions; that is, the @latest [version query](https://go.dev/ref/mod#version-queries) should resolve to the new version before retractions are considered.

---

I'm curious though if v1.0.0 even hit the module caches. If {proxy,sum,index}.golang.org don't know about it, then I wonder whether its worth going through this.

---

https://index.golang.org/index?since=2023-11-06T03:41:28Z&limit=1